metric_config_parser/metric.py

```diff
@@ -164,22 +166,60 @@ def resolve(
     ) -> List[Summary]:
         metric_summary = None
         metric = None
+        upstream_metrics = None
+
+        # check if metric depends on other metrics
+        if self.depends_on:
+            upstream_metrics = []
+            # resolve upstream metrics
+            for metric_ref in self.depends_on:
+                # depends_on chaining of multiple metrics is currently not supported
```

Comment:
Just to clarify, this means `depends_on` depth of 1 only? And we verify this by the below statements looking for an explicit definition of the `upstream_metric` as opposed to more `depends_on` references? Wouldn't a metric with `depends_on` still have a definition?

Reply:
yeah, actually it is supported